cuisines/infrastructure/postgres/cuisines.go

```diff
@@ -0,0 +1,82 @@
+package postgres
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Abuzar-JS/go-spoonacular-api/cuisines/domain"
+
+	"gorm.io/gorm"
+)
+
+type CuisinePostgres struct {
+	db *gorm.DB
+}
+
+func NewCuisinePostgres(db *gorm.DB) *CuisinePostgres {
+	return &CuisinePostgres{
+		db: db,
+	}
+}
+
+type Cuisine struct {
+	ID   int    `gorm:"primaryKey"`
+	Name string `gorm:"column:name;not null;unique"`
+}
+
+func (u Cuisine) TableName() string {
+	return "cuisines"
+}
+
+func (u Cuisine) ToDomain() domain.Cuisine {
+	return domain.Cuisine{
+		ID:   u.ID,
+		Name: u.Name,
+	}
+}
+
+func (u Cuisine) FromDomain(ud domain.Cuisine) Cuisine {
+	return Cuisine{
+		ID:   ud.ID,
+		Name: ud.Name,
+	}
+}
+
+func (u CuisinePostgres) Save(ctx context.Context, request domain.Cuisine) (domain.Cuisine, error) {
+
+	cuisine := Cuisine{}.FromDomain(request)
+
+	result := u.db.Create(&cuisine)
```

Comment:
if result := u.db.Create(&cuisine); result.Error != nil {
		return domain.Cuisine{}, fmt.Errorf("failed to create cuisine")
	}